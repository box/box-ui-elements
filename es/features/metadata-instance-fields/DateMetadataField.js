import * as React from 'react';
import { useIntl } from 'react-intl';
import DatePicker from '../../components/date-picker';
import { convertISOStringToUTCDate } from '../../utils/datetime';
import messages from './messages';
import './DateMetadataField.scss';
const DateMetadataField = ({
  dataKey,
  dataValue,
  displayName,
  description,
  isDisabled,
  onChange,
  onRemove
}) => {
  const {
    formatMessage
  } = useIntl();
  return /*#__PURE__*/React.createElement(DatePicker, {
    className: "bdl-DateMetadataField",
    dateFormat: "utcISOString",
    description: description,
    displayFormat: {
      day: 'numeric',
      month: 'long',
      year: 'numeric'
    },
    hideOptionalLabel: true,
    isDisabled: isDisabled,
    label: displayName,
    onChange: (date, isoDate) => {
      if (isoDate) {
        onChange(dataKey, isoDate);
      } else {
        onRemove(dataKey);
      }
    },
    placeholder: formatMessage(messages.metadataFieldSetDate),
    value: typeof dataValue === 'string' ? convertISOStringToUTCDate(dataValue) : undefined
  });
};
export default DateMetadataField;
//# sourceMappingURL=DateMetadataField.js.map