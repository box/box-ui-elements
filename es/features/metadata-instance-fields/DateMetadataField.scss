@import '../../styles/variables';

.bdl-DateMetadataField {
    .date-picker-icon-holder {
        width: 100%;

        .date-picker-input {
            width: 100%;
        }
    }
}
