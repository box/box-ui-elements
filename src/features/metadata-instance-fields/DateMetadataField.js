// @flow
import * as React from 'react';
import { injectIntl } from 'react-intl';

import DatePicker from '../../components/date-picker';
import { convertISOStringToUTCDate } from '../../utils/datetime';
import messages from './messages';
import type { MetadataFieldValue } from '../../common/types/metadata';
import './DateMetadataField.scss';

type Props = {
    dataKey: string,
    dataValue?: MetadataFieldValue,
    description?: string,
    displayName: string,
    intl: any,
    onChange: (key: string, value: MetadataFieldValue) => void,
    onRemove: (key: string) => void,
};

const DateMetadataField = ({ dataKey, dataValue, displayName, description, intl, onChange, onRemove }: Props) => (
    <DatePicker
        className="bdl-DateMetadataField"
        dateFormat="utcISOString"
        description={description}
        displayFormat={{
            day: 'numeric',
            month: 'long',
            year: 'numeric',
        }}
        hideOptionalLabel
        label={displayName}
        onChange={(date: Date, isoDate: string) => {
            if (isoDate) {
                onChange(dataKey, isoDate);
            } else {
                onRemove(dataKey);
            }
        }}
        placeholder={intl.formatMessage(messages.metadataFieldSetDate)}
        value={typeof dataValue === 'string' ? convertISOStringToUTCDate(dataValue) : undefined}
    />
);

export { DateMetadataField as DateMetadataFieldBase };
export default injectIntl(DateMetadataField);
