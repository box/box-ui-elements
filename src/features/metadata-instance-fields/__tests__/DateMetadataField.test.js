import React from 'react';

import { DateMetadataFieldBase as DateMetadataField } from '../DateMetadataField';

describe('features/metadata-instance-editor/fields/DateMetadataField', () => {
    const intl = {
        formatMessage: jest.fn(),
    };

    test('should correctly render a date field', () => {
        const wrapper = shallow(<DateMetadataField dataValue="2018-06-13T00:00:00.000Z" intl={intl} />);
        expect(wrapper).toMatchSnapshot();
    });
});
