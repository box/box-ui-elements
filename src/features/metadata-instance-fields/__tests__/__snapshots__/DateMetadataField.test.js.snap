// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/metadata-instance-editor/fields/DateMetadataField should correctly render a date field 1`] = `
<DatePicker
  className="bdl-DateMetadataField"
  dateFormat="utcISOString"
  displayFormat={
    Object {
      "day": "numeric",
      "month": "long",
      "year": "numeric",
    }
  }
  hideOptionalLabel={true}
  onChange={[Function]}
  value={2018-06-13T07:00:00.000Z}
/>
`;
